src/Libraries/Microsoft.Extensions.AI/Functions/FromServicesAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Indicates that a parameter to an <see cref="AIFunction"/> should be sourced from an associated <see cref="IServiceProvider"/>.</summary>
+/// <remarks>
+/// <see cref="AIFunctionFactory"/> uses this attribute to determine whether a parameter's value should be sourced from
+/// an <see cref="IServiceProvider"/> instead of from the nominal arguments passed to the function. The <see cref="IServiceProvider"/>
+/// is extracted from the <see cref="AIFunctionArguments.Services"/> property of an <see cref="AIFunctionArguments"/> passed
+/// as the arguments to a call to <see cref="AIFunction.InvokeAsync"/>.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Parameter)]
+public sealed class FromServicesAttribute : Attribute
```

Comment:
I know there's some hesitancy about duplicating the same name MVC uses. I don't have a better idea, though.

Reply:
Do we have further reasons to think it wouldn't suffice to support passing in an `IServiceProvider`, and having the function resolve services from there?

Reply:
Would be worth starting an effort lowering all those attributes to the M.E.DI package? We could then take a dependency on the next preview package.

Reply:
Alternate approaches:
https://github.com/dotnet/extensions/pull/6144
https://github.com/dotnet/extensions/pull/6145
https://github.com/dotnet/extensions/pull/6146

Reply:
> Would be worth starting an effort lowering all those attributes to the M.E.DI package? We could then take a dependency on the next preview package.

Not for a GA release. We wouldn't be able to add that until November. And FromServices is in an ASP.NET MVC namespace.

Reply:
I left a comment on #6146 explaining why I like that proposal the best. Is there any major pushback to making the `IServiceProvider` a top-level parameter like `CancellationToken` to `AIFunction.InvokeAsync`?